src/clusterfuzz/_internal/fuzzing/corpus_manager.py

```diff
@@ -259,12 +280,39 @@ def rsync_from_disk(self,
     """
     corpus_gcs_url = self.get_gcs_url()
     legalize_corpus_files(directory)
-    result = self._gsutil_runner.rsync(directory, corpus_gcs_url, timeout,
-                                       delete)
+    result = self._gsutil_runner.rsync(
+        directory, corpus_gcs_url, timeout, delete=delete)
+
+    # Upload zipcorpus.
+    # TODO(metzman): Get rid of the rest of this function when migration is
+    # complete.
+    filenames = shell.get_files_list(directory)
+    self._upload_to_zipcorpus(filenames, partial=False)
 
     # Allow a small number of files to fail to be synced.
     return _handle_rsync_result(result, max_errors=MAX_SYNC_ERRORS)
 
+  def get_zipcorpora_gcs_urls(self, max_partial_corpora=float('inf')):
+    yield self.get_zipcorpus_gcs_url(partial=False)
+    partial_corpora_gcs_url = f'{self.get_zipcorpus_gcs_dir_url()}/partial*'
+    partial_corpora = storage.list_blobs(partial_corpora_gcs_url)
+    for idx, partial_corpus in enumerate(partial_corpora):
+      if idx > max_partial_corpora:
+        break
+      yield partial_corpus
+
+  def download_zipcorpora(self, dst_dir):
+    zipcorpora_urls = self.get_zipcorpora_gcs_urls()
+    for zipcorpus_url in zipcorpora_urls:
+      # TODO(metzman): Find out what's the tradeoff between writing the file to
+      # disk first or unpacking it in-memory.
+      with get_temp_zip_filename() as temp_zip_filename:
+        if not storage.exists(zipcorpus_url):
```

Comment:
Should we log a warning/error? When would this happen? Race condition from corpus pruning?

Reply:
1. When a fuzzer is new. This is fine and I can't imagine special handling.
2. The pruning race. I think this is pretty rare so I'm going to not handle it specially (one could imagine getting a new list).
I'm going to add logging to keep track of whether 2. is indeed common.